appium/webdriver/webdriver.py

```diff
@@ -341,51 +332,34 @@ def find_elements_by_android_uiautomator(self, uia_string):
         """
         return self.find_elements(by=By.ANDROID_UIAUTOMATOR, value=uia_string)
 
-    def find_element_by_image(self, png_img_path,
-                              match_threshold=DEFAULT_MATCH_THRESHOLD):
+    def find_element_by_image(self, png_img_path):
         """Finds a portion of a screenshot by an image.
         Uses driver.find_image_occurrence under the hood.
 
         :Args:
         - png_img_path - a string corresponding to the path of a PNG image
-        - match_threshold - a double between 0 and 1 below which matches will
-          be rejected as element not found
 
-        :return: an ImageElement object
+        :return: an Element object
         """
-        screenshot = self.get_screenshot_as_base64()
         with open(png_img_path, 'rb') as png_file:
             b64_data = base64.b64encode(png_file.read()).decode('UTF-8')
-        try:
-            res = self.find_image_occurrence(screenshot, b64_data,
-                                             threshold=match_threshold)
-        except WebDriverException as e:
-            if 'Cannot find any occurrences' in str(e):
-                raise NoSuchElementException(e)
-            raise
-        rect = res['rect']
-        return ImageElement(self, rect['x'], rect['y'], rect['width'],
-                            rect['height'])
-
-    def find_elements_by_image(self, png_img_path,
-                               match_threshold=DEFAULT_MATCH_THRESHOLD):
+
+        return self.find_element(by=By.IMAGE, value=b64_data)
+
+    def find_elements_by_image(self, png_img_path):
         """Finds a portion of a screenshot by an image.
         Uses driver.find_image_occurrence under the hood. Note that this will
         only ever return at most one element
 
         :Args:
         - png_img_path - a string corresponding to the path of a PNG image
-        - match_threshold - a double between 0 and 1 below which matches will
-          be rejected as element not found
 
-        :return: possibly-empty list of ImageElements
+        :return: possibly-empty list of Elements
         """
-        els = []
-        try:
-            els.append(self.find_element_by_image(png_img_path, match_threshold))
-        except NoSuchElementException:
-            pass
-        return els
+        with open(png_img_path, 'rb') as png_file:
+            b64_data = base64.b64encode(png_file.read()).decode('UTF-8')
+
+        return self.find_elements(by=By.IMAGE, value=b64_data)
```

Comment:
in java I call this locator type By.imageTemplate. How do you think which one is more appropriate?

Reply:
ah =>
https://github.com/appium/python-client/blob/c6f58bc2f5bfc9cb7ab7ef39ac5b655133f315cf/appium/webdriver/webdriver.py#L111

----

1. As a user, if users should imagine what logic is used behind the method, it's better to use `template` since it help users clalify the behaviour, I think. Then, I'd like to define `-image template` as the selector though.

2. We defined the selector as `-image`. Then, using `image` in both `By` and `Selector` is more appropriate, I felt.

Comparing 1 and 2, so far, `2` is enough, I think. (`By.IMAGE` is much enough.)

(I considered the naming, too 🤔 )